src/TagsInput/components/InputField.js

```diff
@@ -3,72 +3,92 @@
 import * as React from 'react';
 import { View, TextInput, Text } from 'react-native';
 import type { ViewLayoutEvent } from 'react-native/Libraries/Components/View/ViewPropTypes';
+import { defaultTokens } from '@kiwicom/orbit-design-tokens';
 import { StyleSheet } from '../../PlatformStyleSheet';
+import { INPUT_MIN_WIDTH } from '../constants';
 
 type Props = {|
-  +onChangeText: (value: string) => void,
-  +value: string,
-  +onLayout: (e: ViewLayoutEvent) => void,
-  +maxWidth: ?number | 'auto',
-  +minWidth: number,
   +fontSize: number,
-  +disabled: ?boolean,
-  +placeholder: ?string,
-  +forwardRef: (el: any) => void,
-  +onFocus: () => void,
-  +onBlur: () => void,
-  +autoFocus?: boolean,
+  +onChangeText: (value: string) => void,
+  +forwardRef?: (el: any) => void,
+  +minWidth?: number,
+  +value?: string,
+  +onFocus?: () => void,
+  +onBlur?: () => void,
+  +disabled?: boolean,
+  +maxWidth?: number,
+  +placeholder?: string,
+  +onKeyPress?: (e: ViewLayoutEvent) => void,
 |};
 
 type State = {|
-  componentWidth: number | 'auto',
+  componentWidth: number,
 |};
 
-class InputField extends React.PureComponent<Props, State> {
+class InputField extends React.Component<Props, State> {
   static defaultProps = {
-    maxWidth: 'auto',
-    minWidth: 'auto',
+    fontSize: parseFloat(defaultTokens.fontSizeTextLarge),
+    disabled: false,
   };
 
   constructor(props: Props) {
     super(props);
     this.state = {
-      componentWidth: props.minWidth ?? 'auto',
+      componentWidth: props.minWidth ?? INPUT_MIN_WIDTH,
     };
   }
 
-  handleOnLayout = (event: ViewLayoutEvent) => {
-    const { onLayout, maxWidth, minWidth, fontSize } = this.props;
-    const { width: inputWidth } = event.nativeEvent.layout;
+  shouldComponentUpdate(nextProps, nextState) {
+    const { fontSize, maxWidth, minWidth, placeholder, value } = this.props;
+
+    if (
+      fontSize !== nextProps.fontSize ||
```

Comment:
Could you give names or group the conditions to make it more understandable why these conditions are needed?